### ISSUE TITLE
Reproduction of SDAR (Qwen3-1.7B, ALFWorld) underperforms paper by ~8–17 points within 150 steps

### ISSUE THREAD

**Issue by 528why** [ZJU-REAL/SDAR#13]
Hi authors, thanks for releasing SDAR! I tried to reproduce the **Qwen3-1.7B + ALFWorld** result reported in the paper (≈ **53.9%**), but across **5 seeds × 2 independent installs** my best `val/success_rate` within the first 150 steps is consistently 8–17 points below. Would appreciate any pointers.

Metric used below: **best `val/success_rate` observed within the first 150 steps** (`trainer.test_freq=5`, `val_data_size=128`).

---

## 1. Setup

- **Model**: `Qwen/Qwen3-1.7B`; **Task**: ALFWorld (text); **Hardware**: 1 node × 8 × NVIDIA H800.
- **Script**: `examples/sdar_trainer/run_alfworld_qwen3.sh`, **unchanged** (default `sdar_coef=0.01`, `gate_beta=5.0`, `skill_all=false`, `train_data_size=16`, `val_data_size=128`, `group_size=8`, `env.max_steps=50`, `enable_thinking=False`).
- Two independent installs to rule out env-specific issues:

| Env  | Source                                                  | flash-attn               | vllm     |
|------|---------------------------------------------------------|--------------------------|----------|
| Env1 | Custom (added `huggingface_hub<1.0` upper bound)        | `2.8.3+cu12torch2.8`     | `0.11.0` |
| Env2 | **Strictly follows `README.md` Installation section**   | `2.7.4.post1`            | `0.11.0` |

---

## 2. Aggregated results vs. paper

| Env  | # seeds | best `val/success_rate` within 150 steps (mean ± std) | Paper | Gap        |
|------|---------|-------------------------------------------------------|-------|------------|
| Env1 | 3       | **43.49 ± 1.33**                                      | 53.9  | **−10.41** |
| Env2 | 2       | **37.89 ± 1.17**                                      | 53.9  | **−16.01** |

Per-run:

| Env  | wandb run id | best `val/success_rate` (%) | best step |
|------|--------------|-----------------------------|-----------|
| Env1 | `frlo2lzx`   | 42.188                      | 150       |
| Env1 | `urgvw1a3`   | 45.313                      | 125       |
| Env1 | `of1manbx`   | 42.969                      | 150       |
| Env2 | `dhfdrmr1`   | 36.719                      | 125       |
| Env2 | `xugailwz`   | 39.063                      | 150       |

Best single seed is **45.31** (−8.6); 5-seed mean is **41.25** (−12.7).

---

## 3. Per-subtask breakdown vs. paper

Snapshot at the step where `val/success_rate` peaked within 150 steps. Column order matches paper Table 1 / `docs/sdar/metric.png` (Pick / Look / Clean / Heat / Cool / Pick2 / Avg).

| Run                         | Env  | Best step | Pick      | Look      | Clean     | Heat      | Cool      | Pick2     | Avg       |
|-----------------------------|------|-----------|-----------|-----------|-----------|-----------|-----------|-----------|-----------|
| frlo2lzx                    | env1 | 150       | 53.80     | 30.80     | 45.80     | 38.90     | 42.10     | 20.00     | 42.20     |
| urgvw1a3                    | env1 | 125       | 64.70     | 33.30     | 46.70     | 41.20     | 40.00     | 23.80     | 45.30     |
| of1manbx                    | env1 | 150       | 61.80     | 25.00     | 57.70     | 28.60     | 40.00     | 15.80     | 43.00     |
| dhfdrmr1                    | env2 | 125       | 47.10     | 50.00     | 26.70     | 41.20     | 30.00     | 33.30     | 36.70     |
| xugailwz                    | env2 | 150       | 55.88     | 37.50     | 42.31     | 19.05     | 40.00     | 26.32     | 39.06     |
| **My mean (5 seeds)**       |      |           | **56.66** | **35.32** | **43.84** | **33.79** | **38.42** | **23.85** | **41.25** |
| **Paper SDAR (Qwen3-1.7B)** | —    | —         | **73.5**  | **25.0**  | **76.9**  | **30.0**  | **50.0**  | **36.8**  | **53.9**  |
| **Gap (paper − mine)**      |      |           | **−16.8** | **+10.3** | **−33.1** | **+3.8**  | **−11.6** | **−13.0** | **−12.6** |

---

## 4. Sanity checks (already ruled out)

- Both envs (custom + strict-README) underperform; Env2 is even worse than Env1.
- `val_before_train=True` returns ~12.5% (matches zero-shot Qwen3-1.7B) → eval pipeline is correct.
- `enable_thinking=False` is set; all README-mandated package versions verified post-install.
- Dataset is the canonical `alfworld-download -f` 2.1.1 set; `~/data/verl-agent/text/{train,test}.parquet` from `examples.data_preprocess.prepare --mode text`.

---

## 5. Questions for the authors

1. **Why does the reproduction look so far behind within the first 150 steps?** 5-seed mean is **41.25%** (best 45.31%), vs. paper's 53.9. The per-subtask gap is strongly skewed (Clean −33, Pick −17, Pick2 −13, Cool −12, but Look +10, Heat +4), which doesn't look like generic "under-training". Is there anything in the default `run_alfworld_qwen3.sh` config, the `skills/alfworld/` skill set, or the dependency stack from `pip install -e . + vllm==0.11.0 + flash-attn==2.7.4.post1` that you would expect to materially change the 150-step number? Knowing roughly where your training curve sits at step 150 would already help a lot.

2. **Continuing past 150 steps lifts `val/success_rate` to 52.3% / 60.9% at step 300, clearly above the reported 53.9 — was there a specific reason to stop at ~150 steps and not report the longer-horizon numbers?** Two Env1 runs trained past 150 steps and reached `val/success_rate` of **52.3%** (`urgvw1a3`) and **60.9%** (`of1manbx`) at step 300. If longer training reliably yields a stronger model, then 53.9 is arguably a conservative estimate of SDAR's actual capability. Was there a particular consideration behind stopping around 150 steps (e.g. compute budget, alignment with baselines' training length, instability when trained longer)?

Happy to share full `output.log`, `wandb-summary.json`, `requirements.txt` for any of the 5 runs. Thanks again!

**Comment by lll6gg** [comment]
<img width="2182" height="1430" alt="Image" src="https://github.com/user-attachments/assets/06414d22-0f08-4875-8b1a-cb898091cb24" />

Thanks a lot for the comments.

A few comments that may explain the gap:

1. The ~150 training steps were mainly to align with the Qwen2.5-series experiments, since Qwen2.5 models usually saturate around that region. We kept the same budget for consistency across models.
2. Qwen3-1.7B does not seem to saturate at 150 steps. We have uploaded our validation curves, and you can see a noticeable jump at ~150 steps, with the curve still improving afterward. So we agree that Qwen3-1.7B likely needs more training steps.
3. We suspect the main reason is that Qwen3-1.7B is less stable than the Qwen2.5 series under the current GRPO+OPSD setup, and the gains are sensitive. For reproduction, we currently recommend starting from the more stable Qwen2.5, and then using longer schedules for Qwen3-1.7B.

It would also be great if you could share the full validation curves. Thanks again for the careful effort!

**Comment by 528why** [comment]
Here are the complete validation curves for env1 and env2:

<img width="1570" height="755" alt="Image" src="https://github.com/user-attachments/assets/59b517f0-b240-4c01-b108-7ad27441b721" />
<div align="center">env1</div>


<img width="1560" height="752" alt="Image" src="https://github.com/user-attachments/assets/90e0995c-5f12-4e1f-94e4-a729a6d27318" /> 
<div align="center">env2</div>

**Comment by lll6gg** [comment]
Thanks, env1 seems closer to my results.

I have tried 2 times for training, both the results are around 0.50. I doubt this is caused by enviroment instability.
